src/Inventory.ts

```diff
@@ -125,26 +118,19 @@ export function retrieveTools (bot: Bot, options: ToolRetrievalOptions, cb: Call
  * @param cb - The callback to run when finished.
  */
 function gotoChest (bot: Bot, location: Vec3, cb: Callback): void {
-  // @ts-expect-error
   const pathfinder = bot.pathfinder
 
   pathfinder.setGoal(new goals.GoalBlock(location.x, location.y, location.z))
 
-  const events = new TemporarySubscriber(bot)
-  events.subscribeTo('goal_reached', () => {
-    events.cleanup()
-    cb()
-  })
+  bot.once('goal_reached', cb)
 
-  events.subscribeTo('path_update', (results: ComputedPath) => {
+  bot.once('path_update', (results: ComputedPath) => {
```

Comment:
`bot.once` cannot be used here as `path_update` is not guaranteed to be called only once. (May be called many times before being removed.)